language/api/Analyze/Program.cs

```diff
@@ -210,27 +210,43 @@ private static void WriteEntitySentiment(IEnumerable<Entity> entities)
         // [END analyze_entity_sentiment_from_file]
         // [END analyze_entity_sentiment_from_string]
 
-        // [START analyze_entities_from_file]
-        private static void AnalyzeEverything(string text)
+        private static void AnalyzeEverything(string text,
+            bool isEntitySentimentAvailable)
         {
             var client = LanguageServiceClient.Create();
-            var response = client.AnnotateText(new Document()
+            try
             {
-                Content = text,
-                Type = Document.Types.Type.PlainText
-            },
-            new Features()
+                var response = client.AnnotateText(new Document()
+                {
+                    Content = text,
+                    Type = Document.Types.Type.PlainText
+                },
+                new Features()
+                {
+                    ExtractSyntax = true,
+                    ExtractDocumentSentiment = true,
+                    ExtractEntities = true,
+                    ExtractEntitySentiment = isEntitySentimentAvailable
+                });
+                Console.WriteLine($"Language: {response.Language}");
+                WriteSentiment(response.DocumentSentiment, response.Sentences);
+                WriteSentences(response.Sentences, response.Tokens);
+                WriteEntities(response.Entities);
+                if (isEntitySentimentAvailable)
+                {
+                    WriteEntitySentiment(response.Entities);
+                }
+                else
+                {
+                    Console.WriteLine("Entity Sentiment: Non-English language "
+                        + "detected. While this feature is currently in Beta "
+                        + "only English is supported.");
+                }
+            }
+            catch (Grpc.Core.RpcException)
             {
-                ExtractSyntax = true,
-                ExtractDocumentSentiment = true,
-                ExtractEntities = true,
-                ExtractEntitySentiment = true
-            });
-            Console.WriteLine($"Language: {response.Language}");
-            WriteSentiment(response.DocumentSentiment, response.Sentences);
-            WriteSentences(response.Sentences, response.Tokens);
-            WriteEntities(response.Entities);
-            WriteEntitySentiment(response.Entities);
+                AnalyzeEverything(text, false);

```

Comment:
If there's any problem *other* than sentiment analysis not being available, this is going to recurse forever. Basically we're assuming one particular cause, where there could be any number of real causes.